### PR TITLE
fix: deprecate goerli network

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

Pr to deprecate goerli network. 
- Removes instances of goerli network. (From network dropdown and setting)
- Updates e2e tests to use Sepolia instead.
- Creates a new component (https://www.figma.com/file/KXgbxFucvk3FM5VULGZmzk/Assets%3A-Bug-Fixes?node-id=449%3A2210&mode=dev) for the scenario when a user clicks on a deepling for goerli.

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/1368

## **Manual testing steps**

1. Click on network dropdown list (you should not see Goerli)
2. Click on settings, then network list (you should not see Goerli)
3. Click on a deeplink that is supposed to switch to goerli : https://metamask.app.link/send/0x4701Aa9471d7bfAc765D87dcb1Ea6BB23AD32733@5/approve?address=0x178e3e6c9f547A00E33150F7104427ea02cfc747&uint256=5e8. You should see a modal saying that the network is deprecated.

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/10994169/d748b713-5157-4d00-8b85-558c1e0164a7

### **After**

When you try to click on a deeplink with goerli

https://github.com/MetaMask/metamask-mobile/assets/10994169/622a47c3-5223-4420-84d8-a2aa3274973d

When you try to add goerli network as custom network

https://github.com/MetaMask/metamask-mobile/assets/10994169/0591d2b9-3299-44aa-9e6b-463ff66cd693

Android:

https://github.com/MetaMask/metamask-mobile/assets/10994169/13f4ef7f-adcc-4967-8036-1ceffbf0653a

https://github.com/MetaMask/metamask-mobile/assets/10994169/aa6028b1-5ab0-4063-b00f-6c47f941880f

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/store/migrations/029.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export default function migrate(state) {
+  const chainId =
+    state.engine.backgroundState.NetworkController.providerConfig.chainId;
+  // Deprecate rinkeby, ropsten and Kovan, goerli any user that is on those we fallback to sepolia
+  if (
+    chainId === '4' ||
+    chainId === '3' ||
+    chainId === '42' ||
+    chainId === '5'
```
Comment (sahar-fehri) [inline app/store/migrations/029.js:12]: Im not quite sure if i created this migration correctly but please lemme know 🙏

File: e2e/specs/swaps/token-details.spec.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ describe(SmokeSwaps('Token Chart Tests'), () => {
   });
 
   it('should not display the chart when using Goerli test network', async () => {
-    await switchToGoreliNetwork();
+    await switchToSepoliaNetwork();
     await WalletView.tapOnToken(Networks.Goerli.providerConfig.ticker);
```
Comment (cortisiko) [inline e2e/specs/swaps/token-details.spec.js:36]: Update this to reflect Sepolia

File: e2e/specs/assets/import-nft.spec.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { importWalletWithRecoveryPhrase } from '../../viewHelper';
 import Collectibles from '../../resources/collectibles.json';
 
 describe(SmokeCore('Import NFT'), () => {
```
Comment (cortisiko) [inline e2e/specs/assets/import-nft.spec.js:11]: Can you update this test to use fixtures to deploy a 721 contract and then import it into the wallet? 

You can use https://github.com/MetaMask/metamask-extension/blob/develop/test/e2e/tests/nft/import-nft.spec.js as a guide.

**Review by cortisiko — CHANGES_REQUESTED** [review]
Left a few comments relating to E2E

**Review by cortisiko — CHANGES_REQUESTED** [review]

File: e2e/specs/assets/import-nft.spec.js [hunk 4]
```diff
@@ -20,13 +25,13 @@ describe(SmokeCore('Import NFT'), () => {
     await importWalletWithRecoveryPhrase();
   });
 
-  it('should switch to Goerli network', async () => {
+  it('should switch to Sepolia network', async () => {
```
[the remark this answers is not in the document]
Comment (sahar-fehri) [reply e2e/specs/assets/import-nft.spec.js]: Thanks Curtis 🙏  fixed that 🙏

**Review by cortisiko — DISMISSED** [review]
From a testing perspective this PR is ✅

**Review by tommasini — CHANGES_REQUESTED** [review]
As requested Sahar, I took a look on the migration file, and left some question and outputs how we could handle this on the best way possible!

Let me know any support that you need for we further adopt this migration process, anything just ping me!

File: app/store/migrations/029.js (hunk 1 above)
Comment (sahar-fehri) [reply app/store/migrations/029.js]: Thanks a lot Tomas for the feedback! 🙏  will update this 🙏

**Review by tommasini — CHANGES_REQUESTED** [review]
Awesome work! I left some comments

File: app/core/DeeplinkManager/Handlers/handleEthereumUrl.ts [hunk 5]
```diff
@@ -27,6 +28,12 @@ async function handleEthereumUrl({
   const txMeta = { ...ethUrl, source: url };
 
   try {
+    // If the deeplink has a goerli chainId, show deprecation modal and return
+    if (ethUrl.chain_id === getDecimalChainId(CHAIN_IDS.GOERLI)) {
```
[the remark this answers is not in the document]
Comment (sahar-fehri) [reply app/core/DeeplinkManager/Handlers/handleEthereumUrl.ts]: Nice catch! Fixed that 🙏

File: app/components/UI/DeprecatedNetworkModal/DeprecatedNetworkModal.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    Linking.openURL(CONNECTING_TO_DEPRECATED_NETOWRK);
+    AnalyticsV2.trackEvent('EXTERNAL_LINK_CLICKED', {
+      location: 'dapp_connection_request',
+      text: 'Learn More',
+      url_domain: CONNECTING_TO_DEPRECATED_NETOWRK,
+    });
+  };
+
+  return (
+    <ReusableModal ref={modalRef} style={styles.screen}>
```
[the remark this answers is not in the document]
Comment (sahar-fehri) [reply app/components/UI/DeprecatedNetworkModal/DeprecatedNetworkModal.tsx]: Thats true, i ve missed that component 🙏  I updated the file, the BottomSheetHeader was not looking like i want so i used the Text instead for the title

**Review by Cal-L — COMMENTED** [review]
Left some comments

**Review by Cal-L — COMMENTED** [review]
Have we also tested upgrading with Goerli network selected?

**Comment by sahar-fehri** [comment]
> Have we also tested upgrading with Goerli network selected?

Hey @Cal-L , i have tested the migration when the user is already on goerli, he will be automatically switched to Sepolia 🙏

File: app/components/UI/DeprecatedNetworkModal/DeprecatedNetworkModal.styles.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    button: {
+      flex: 1,
+      marginLeft: 16,
+      marginRight: 16,
+      marginTop: 24,
+      marginBottom: Device.isAndroid() ? 21 : 0,
+    },
+    buttonLabel: {
+      fontSize: 14,
```
[the remark this answers is not in the document]
Comment (sahar-fehri) [reply app/components/UI/DeprecatedNetworkModal/DeprecatedNetworkModal.styles.ts]: Thnx for pointing this out! fixed this

**Review by Cal-L — COMMENTED** [review]
Code LGTM, let's resolve the conflict and I'll approve

**Review by Cal-L — DISMISSED** [review]
LGTM.

**Review by tommasini — APPROVED** [review]
LGTM

**Pull request merged**